plugin/storage/factory.go

```diff
@@ -137,8 +137,16 @@ func (*Factory) getFactoryOfType(factoryType string) (storage.Factory, error) {
 // Initialize implements storage.Factory.
 func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
 	f.metricsFactory = metricsFactory
-	for _, factory := range f.factories {
-		if err := factory.Initialize(metricsFactory, logger); err != nil {
+	for name, factory := range f.factories {
+		if err := factory.Initialize(
+			metricsFactory.Namespace(metrics.NSOptions{
+				Name: "storage",
+				Tags: map[string]string{
+					"name": name,
+					// TODO: how should we get kind?
```

Comment:
`for name, factory` is actually `for kind, factory`, because in v1 there is notion of storage names. The closest v1 has to names is primary/archive, which you might add to the namespace as a name attribute in CreateReader vs. CreateArchiveReader

Reply:
@yurishkuro We're not recording any metrics today for the archive span reader so can we just leave the name tag off?